internal/errors/transport.go

```diff
@@ -122,3 +122,15 @@ func (e ProcedureFailedError) Error() string {
 func (e ProcedureFailedError) AsHandlerError() HandlerError {
 	return HandlerUnexpectedError(e)
 }
+
+// UnknownRPCModeError is a failure to process a request because the RPC
+// mode is unkown to the transport
+//TODO: add tests for this error
+type UnknownRPCModeError struct {
```

Comment:
Let's refer to this concept as `RPC Type` - this is consistent with gRPC's terminology:

> First let’s look at the simplest type of RPC, where the client sends a single request and gets back a single response.

from http://www.grpc.io/docs/guides/concepts.html#unary-rpc

Suggest renaming this to `UnkownRPCTypeError` and always referring to this concept as `RPC Type` going forward.